lib/session.js

```diff
@@ -108,6 +109,9 @@ var doSignin = function(lookup, req, res, onSuccess, onFail) {
 			if (user && (!passwordCheck || scmp(passwordCheck, hash(user.password)))) {
 				postHookedSigninWithUser(user, req, res, onSuccess, onFail);
 			} else {
+				if (!err) {
+					err = new Error('The password you entered is incorrect.');
```

Comment:
That's true. But in general, at least in my opinion, it is always a bad idea to state that the password is incorrect. It implies that the username was correct which is bad for two reasons:
1) The username may still be wrong
2) It may leak information about the existing user accounts and give better opportunities to hackers